packages/protocol/contracts/stability/Reserve.sol

```diff
@@ -394,6 +423,75 @@ contract Reserve is
     return _transferGold(to, value);
   }
 
+  /**
+   * @notice Transfer ERC20 token to a whitelisted address subject to reserve spending limits is the limit is set, othersise the limit is 100%.
+   * @param to The address that will receive the ERC20 token.
+   * @param value The amount of ERC20 token to transfer.
+   * @param erc20TokenAddress The token address you're transferring.
+   * @return Returns true if the transaction succeeds.
+   */
+  function transferErc20Token(address erc20TokenAddress, address payable to, uint256 value)
+    external
+    returns (bool)
+  {
+    require(
+      erc20TokenAddress != registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID),
+      "token address should not equal to celo"
+    );
+    require(isSpender[msg.sender], "sender not allowed to transfer Reserve funds");
+    require(isOtherReserveAddress[to], "can only transfer to other reserve address");
+    uint256 spendingLimitForThisToken;
+    if (FixidityLib.unwrap(erc20DailySpendingRatio[erc20TokenAddress]) > 0) {
+      uint256 currentDay = now / 1 days;
+      if (currentDay > lastSpendingDay) {
+        uint256 balance = getErc20TokenBalance(erc20TokenAddress);
+        lastSpendingDay = currentDay;
+        spendingLimitForThisToken = erc20DailySpendingRatio[erc20TokenAddress]
+          .multiply(FixidityLib.newFixed(balance))
+          .fromFixed();
+      }
+      require(spendingLimitForThisToken >= value, "Exceeding spending limit");
+      spendingLimitForThisToken = spendingLimitForThisToken.sub(value);
+      return _transferErc20Token(to, value, erc20TokenAddress);
+    } else {
+      return _transferErc20Token(to, value, erc20TokenAddress);
+    }
+  }
+
+  /**
+   * @notice Transfer unfrozen ERC20 token to any address.
```

Comment:
In general, I would suggest we remove the entire frozen CELO logic. We do not need it anymore at this point as it was just needed for kickstarting the reserve. We are at 100% unfrozen and I do not see a scenario in which we would want to freeze parts again. Could you try to remove the differentiation between frozen and unfrozen entirely to simplify the setup?